pulsar/consumer_test.go

```diff
@@ -384,6 +385,138 @@ func TestPartitionTopicsConsumerPubSub(t *testing.T) {
 	assert.Equal(t, len(msgs), 10)
 }
 
+type TestActiveConsumerListener struct {
+	lock             sync.RWMutex
+	nameToPartitions map[string]map[int32]struct{}
+}
+
+func (t *TestActiveConsumerListener) getConsumerCount() int {
+	t.lock.RLock()
+	defer t.lock.RUnlock()
+	return len(t.nameToPartitions)
+}
+
+func (t *TestActiveConsumerListener) getPartitionCount(consumerName string) int {
+	t.lock.RLock()
+	defer t.lock.RUnlock()
+	return len(t.nameToPartitions[consumerName])
+}
+
+func (t *TestActiveConsumerListener) BecameActive(consumer Consumer, topicName string, partition int32) {
+	fmt.Printf("%s become active on %s - %d\n", consumer.Name(), topicName, partition)
+	t.lock.Lock()
+	defer t.lock.Unlock()
+	partitionSet := t.nameToPartitions[consumer.Name()]
+	if partitionSet == nil {
+		partitionSet = map[int32]struct{}{}
+	}
+	partitionSet[partition] = struct{}{}
+	t.nameToPartitions[consumer.Name()] = partitionSet
+}
+
+func (t *TestActiveConsumerListener) BecameInactive(consumer Consumer, topicName string, partition int32) {
+	fmt.Printf("%s become inactive on %s - %d\n", consumer.Name(), topicName, partition)
+	t.lock.Lock()
+	defer t.lock.Unlock()
+	partitionSet := t.nameToPartitions[consumer.Name()]
+	if _, ok := partitionSet[partition]; ok {
+		delete(partitionSet, partition)
+		if len(partitionSet) == 0 {
+			delete(t.nameToPartitions, consumer.Name())
+		}
+	}
+}
+
+func allConsume(consumers []Consumer) {
+	for i := 0; i < len(consumers); i++ {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+		consumers[i].Receive(ctx)
+	}
+}
+
+func TestPartitionTopic_ActiveConsumerChanged(t *testing.T) {
+	client, err := NewClient(ClientOptions{
+		URL: lookupURL,
+	})
+	assert.Nil(t, err)
+	defer client.Close()
+
+	topic := "persistent://public/default/testGetPartitions6"
```

Comment:
Better to add timestamp as the topic name suffix. Like `fmt.Sprintf("my-topic-%v", time.Now().Nanosecond())` to avoid conflict when running multiple times. You can also use `newTopicName` in helper_for_test.go

Reply:
Great, I've change it!